src/utils/conversions.rs

```diff
@@ -1,127 +1,182 @@
+            'F' => Temperature::from_fahrenheit(temp[..temp.len() - 1].parse::<f64>()?),
+            'C' => Temperature::from_celsius(temp[..temp.len() - 1].parse::<f64>()?),
+            'K' => Temperature::from_kelvin(temp[..temp.len() - 1].parse::<f64>()?),
+            'R' => Temperature::from_rankine(temp[..temp.len() - 1].parse::<f64>()?),
+            _ => Temperature::from_celsius(temp.parse::<f64>()?),
         };
-        ret
+        Ok(ret)
     }
 }
 
 pub struct VolumeBuilder;
 
 impl VolumeBuilder {
-    pub fn from_str<S: Into<String>>(val: S) -> measurements::Volume {
+    pub fn from_str<S: Into<String>>(val: S) -> Result<measurements::Volume, ParseFloatError> {
         let vol = val.into().to_lowercase();
         if vol.is_empty() {
-            return Volume::from_litres(0.0);
+            return Ok(Volume::from_litres(0.0));
+        }
+
+        match vol.find(':') {
+            Some(count) => {
+                let (temperature, temperature_form) = vol.split_at(count);
+                println!("{}", temperature);
+                println!("{}", temperature_form);
+            }
+            None => {}
         }
 
         if vol.len() > 3 {
             let last3 = &vol[vol.len() - 3..];
+            println!("{}", last3);
+            println!("{}", &vol[..vol.len() - 3]);
             let ret = match last3 {
-                "cm3" => Ok(Volume::from_cubic_centimeters(vol[..vol.len()-3].parse::<f64>().unwrap())),
-                "ft3" => Ok(Volume::from_cubic_feet(vol[..vol.len()-3].parse::<f64>().unwrap())),
-                "yd3" => Ok(Volume::from_cubic_yards(vol[..vol.len()-3].parse::<f64>().unwrap())),
-                "in3" => Ok(Volume::from_cubic_inches(vol[..vol.len()-3].parse::<f64>().unwrap())),
-                "gal" => Ok(Volume::from_gallons(vol[..vol.len()-3].parse::<f64>().unwrap())),
-                "cup" => Ok(Volume::from_cups(vol[..vol.len()-3].parse::<f64>().unwrap())),
-                "tsp" => Ok(Volume::from_teaspoons(vol[..vol.len()-3].parse::<f64>().unwrap())),
-                _ => Err(last3)
+                "cm3" => Volume::from_cubic_centimeters(vol[..vol.len() - 3].parse::<f64>()?),
+                "ft3" => Volume::from_cubic_feet(vol[..vol.len() - 3].parse::<f64>()?),
+                "yd3" => Volume::from_cubic_yards(vol[..vol.len() - 3].parse::<f64>()?),
+                "in3" => Volume::from_cubic_inches(vol[..vol.len() - 3].parse::<f64>()?),
+                "gal" => Volume::from_gallons(vol[..vol.len() - 3].parse::<f64>()?),
+                "cup" => Volume::from_cups(vol[..vol.len() - 3].parse::<f64>()?),
+                "tsp" => Volume::from_teaspoons(vol[..vol.len() - 3].parse::<f64>()?),
+                _ => Volume::from_cubic_centimeters(vol[..vol.len() - 3].parse::<f64>()?),
             };
-
-            if ret.is_ok() {
-                return ret.unwrap();
-            }
-        }
-
-        if vol.len() > 2 {
+            Ok(ret)
+        } else if vol.len() > 2 {
```

Comment:
I think @drodil had a reason not having them mutually exclusive. One reason comes to mind, if `len` > 2 but the suffix isn't one of `ml`, `m3`, `ul`, or `dr`.

Reply:
I've made a different system that's far from perfect, but will parse correctly-the previous one could fail due to lengths of the string while parsing with lower length formats (For example "123ml" would fail, as the length is > 3, so it'd attempt to parse as a three lettered suffix, leading to the return being 12 cubic centimeters, instead of 123 milliliters)

Reply:
I think if it can't find `cm3`, it will try again for shorter suffixes, and `ml` will eventually match. The current implementation might not be perfect, but these changes add some unnecessary verbosity to the required inputs, also requires calls to be `unwrap`.